### ISSUE TITLE
DefaultJWTService createBaseJWT throws a NullPointerException

### ISSUE THREAD

**Issue by johnmaia** [FusionAuth/fusionauth-issues#399]
### Description 
After updating `fusionauth` version from `1.10.1` to `1.11.0`, after running flawlessly for the last 13 days, during a `exchangeRefreshTokenForJWT` (`POST /api/jwt/refresh`)  a `NullPointerException` was thrown by the `DefaultJWTService` `createBaseJWT` method.

### Steps to reproduce
1. Run `fusionauth` `v1.11.0`
2. After a few hours of normal usage...
3. Request `exchangeRefreshTokenForJWT`
4. The following `NullPointerException` is thrown:

```
Dec 09, 2019 7:03:14.882 PM ERROR io.fusionauth.app.primeframework.error.ExceptionExceptionHandler - An unhandled exception was thrown
java.lang.NullPointerException: null
    at io.fusionauth.api.service.jwt.DefaultJWTService.createBaseJWT(DefaultJWTService.java:411)
    at io.fusionauth.api.service.jwt.DefaultJWTService.refreshAccessToken(DefaultJWTService.java:162)
    at io.fusionauth.app.action.api.jwt.RefreshAction.post(RefreshAction.java:140)
    at sun.reflect.GeneratedMethodAccessor92.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.primeframework.mvc.util.ReflectionUtils.invoke(ReflectionUtils.java:414)
    at org.primeframework.mvc.action.DefaultActionInvocationWorkflow.execute(DefaultActionInvocationWorkflow.java:79)
    at org.primeframework.mvc.action.DefaultActionInvocationWorkflow.perform(DefaultActionInvocationWorkflow.java:62)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.validation.DefaultValidationWorkflow.perform(DefaultValidationWorkflow.java:47)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.security.DefaultSecurityWorkflow.perform(DefaultSecurityWorkflow.java:65)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.parameter.DefaultPostParameterWorkflow.perform(DefaultPostParameterWorkflow.java:50)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.content.DefaultContentWorkflow.perform(DefaultContentWorkflow.java:52)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.parameter.DefaultParameterWorkflow.perform(DefaultParameterWorkflow.java:57)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.parameter.DefaultURIParameterWorkflow.perform(DefaultURIParameterWorkflow.java:102)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.scope.DefaultScopeRetrievalWorkflow.perform(DefaultScopeRetrievalWorkflow.java:58)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.message.DefaultMessageWorkflow.perform(DefaultMessageWorkflow.java:45)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at io.fusionauth.app.primeframework.FrontEndTenantWorkflow.perform(FrontEndTenantWorkflow.java:69)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.action.DefaultActionMappingWorkflow.perform(DefaultActionMappingWorkflow.java:126)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.workflow.StaticResourceWorkflow.perform(StaticResourceWorkflow.java:97)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.parameter.RequestBodyWorkflow.perform(RequestBodyWorkflow.java:89)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at org.primeframework.mvc.security.DefaultSavedRequestWorkflow.perform(DefaultSavedRequestWorkflow.java:57)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at io.fusionauth.app.primeframework.CORSFilter.handleNonCORS(CORSFilter.java:289)
    at io.fusionauth.app.primeframework.CORSFilter.doFilter(CORSFilter.java:250)
    at io.fusionauth.app.primeframework.CORSRequestWorkflow.perform(CORSRequestWorkflow.java:48)
    at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43)
    at io.fusionauth.app.primeframework.FusionAuthMVCWorkflow.perform(FusionAuthMVCWorkflow.java:88)
    at org.primeframework.mvc.workflow.DefaultWorkflowChain.continueWorkflow(DefaultWorkflowChain.java:44)
    at org.primeframework.mvc.servlet.FilterWorkflowChain.continueWorkflow(FilterWorkflowChain.java:50)
    at org.primeframework.mvc.servlet.PrimeFilter.doFilter(PrimeFilter.java:84)
    at com.inversoft.maintenance.servlet.MaintenanceModePrimeFilter.doFilter(MaintenanceModePrimeFilter.java:59)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at com.inversoft.servlet.UTF8Filter.doFilter(UTF8Filter.java:27)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
    at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)
    at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
    at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:808)
    at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
    at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    at java.lang.Thread.run(Thread.java:748)
```

### Platform
- [`fusionauth/fusionauth-app:1.11.0`](https://hub.docker.com/layers/fusionauth/fusionauth-app/1.11.0/images/sha256-20007a57a8701f7bd7d51e865cadfe50590076fbcfcd02e8d35649ce404af6e9) docker image

### Additional context
Fusionauth memory is locked at 2GB.
```
FUSIONAUTH_MEMORY=2048
```

I'll try to gather more information on how an why this error occurred to help you mitigate this bug.wow

**Comment by robotdan** [comment]
Which client library are you using with `exchangeRefreshTokenForJWT` ?

**Comment by johnmaia** [comment]
> A couple of questions:
> 
>     * Which client library are you using with `exchangeRefreshTokenForJWT` ?

The [`fusionauth-node-client@1.11.0`](https://github.com/FusionAuth/fusionauth-node-client/releases/tag/1.11.0).

>     * How many tenants do you have?

In this particular environment we have around 50 tenants. 

>     * To confirm, you've been running ok at 1.11.0 for 13 days, and then this started to occur?

Yes, but it only occurred once. And we are using the same fusionauth version on several other environments where this has never occurred. 

>     * What is your expiration set to for refresh tokens? (may be different per tenant, or per application)?

The expiration time for the refresh tokens on every application is set to 2 days (default).

>     * Is this happening in all tenants and applications, just one, or several?

This only happened once.

> In theory this should be impossible. This could occur if the user were to not exist, but since the refresh token has a FK to the user, that would be impossible.
> 
> There is currently a very small timing window where the user could be deleted after we've retrieved and validated the refresh token and then (in theory) this exception could occur. This seems extremely unlikely unless you're batch deleting users at the same time you're using the Refresh JWT API, even then, seems unlikely.

We are not deleting the users on any given moment on fusionauth.

**Comment by robotdan** [comment]
Thanks for the answers @johnmaia, I can't really explain this yet. If it happens again let me know.

**Comment by johnmaia** [comment]
Hello there @robotdan! 

We've found out what was causing this issue and I can now tell you, with some grade of certainty, that it was a bug on our front-end application that was using a refresh on a application from tenant `A` but it actually belonged to an application from tenant `B`.

Nonetheless, I was expecting an `404` or a `400` error instead of a `500` from fusionauth.

**Comment by robotdan** [comment]
Thanks for the update @johnmaia - sounds like a bug on both sides! :-) Yes, in that case we should have returned a `404` or a validation message with a `400` response. 

If I can recreate based upon this information we can ensure we are returning a more helpful status and message. 

Thanks!

**Comment by matthew-altman** [comment]
Reproduced in 1.15.2
Will be fixed in 1.15.3
